.ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh

```diff
@@ -23,7 +23,7 @@ pushd $local_path
 echo "Checking for modified go files"
 # get the names of changed files and look for go files
 # (ignoring "no matches found" errors from grep)
-gofiles=$(git diff --name-only HEAD~1 | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" || test $? = 1; })
```

Comment:
I think this should fix the go file detection issue occurs when no diff generated for the google providers

Reply:
I can't get this to run locally so I'm a little hesitant about including it. If it works in cloudbuild then that's fine for now at least, though.

Reply:
I'll do a more thorough test on this feature. I'll exclude it from this PR and create a separate PR for it since I don't think feature branch is blocked without this change.